src/lpython/parser/tokenizer.re

```diff
@@ -67,27 +75,23 @@ void lex_int(Allocator &al, const unsigned char *s,
         prefix.p = (char*) "Hex";
         prefix.n = 3;
         Str num;
-        num.p = (char*)s;
+        num.p = get_value(al, (char*)s, 16);
         num.n = e-s;
         u.from_largeint(al, num);
     } else if (std::tolower(s[1]) == 'b') {
         s = s + 2;
         prefix.p = (char*) "Bin";
         prefix.n = 3;
         Str num;
-        num.p = (char*)s;
+        num.p = get_value(al, (char*)s, 2);
         num.n = e-s;
         u.from_largeint(al, num);
     } else if ((std::tolower(s[1]) == 'o')) {
-        if(std::isdigit(s[1])) {
-            s++;
-        } else {
-            s = s + 2;
-        }
+        s = s + 2;
         prefix.p = (char*) "Oct";
         prefix.n = 3;
         Str num;
-        num.p = (char*)s;
+        num.p = get_value(al, (char*)s, 8);
```

Comment:
That's fine for now, but it's not very efficient. We should convert directly without an intermediate string.